functionaltests/utils_test.go

```diff
@@ -32,18 +33,15 @@ import (
 	"github.com/elastic/apm-server/functionaltests/internal/terraform"
 )
 
-// ecAPICheck verifies if EC_API_KEY env var is set.
-// This is a simple check to alert users if this necessary env var
-// is not available.
-//
-// Functional tests are expected to run Terraform code to operate
-// on infrastructure required for each test and to query Elastic
-// Cloud APIs. In both cases a valid API key is required.
-func ecAPICheck(t *testing.T) {
+// getLatestSnapshotFor retrieves the latest snapshot version for the version prefix.
+func getLatestSnapshotFor(t *testing.T, prefix string) string {
 	t.Helper()
-	require.NotEmpty(t, os.Getenv("EC_API_KEY"), "EC_API_KEY env var not set")
+	version, ok := fetchedSnapshots.LatestFor(prefix)
+	require.True(t, ok, "no version with prefix '%s' found", prefix)
```

Comment:
suggestion: could you clarify in the message where this version is missing from? (My understanding is that versions are region specific). Es :

```suggestion
	require.True(t, ok, "no version with prefix '%s' found in Elastic Cloud region %s", prefix, regionFrom(*environment))
```

Reply:
Done in [9c3bab4](https://github.com/elastic/apm-server/pull/16295/commits/9c3bab43bb6ad82388293f8fae8a41a54054d590).